### ISSUE TITLE
Allow kinit and kadmin to get password via file

### ISSUE THREAD

**Issue by macrotex** [heimdal/heimdal#530]
Some use cases would benefit from `kinit` and `kadmin` allowing user input of a password via file. One such case we have is changing a principal's password via web form. While this _can_ be accomplished with tools like Expect, that route is clumsy. The user interface could be something like

```kadmin -l --password-file=/path/to/file cpw username```

`kadmin` would take the first line of `/path/to/file` and use that as the password.

**Comment by jaltman** [comment]
Why a password stored in a file instead of a keytab?

**Comment by macrotex** [comment]
We have some integrations from other servers that enable setting password through a web interface running on another server. This enables us to add additional security to the password change (e.g., two-step). This requires us to pass the password to be changed from this web server to the KDC server to make the change by scripts on the KDC server.

**Comment by jaltman** [comment]
Whose password is in the proposed file?   Based upon the example you provided it would be the password of the Kerberos administrator and not new password for the "cpw --password=<new-pw-string> <principal>" command.

**Comment by macrotex** [comment]
No, it is the password of the user. For example, to change the password for principal `johndoe` using kadmin you do something like this: `kadmin -p admin-principal --keytab /path/to/keytab cpw johndoe`. You then have to type in the password twice. I need something like `--password-file=/path/to/file` to do the password setting rather than having an actual person (or Expect) do the typing in for you.

**Comment by jaltman** [comment]
The password for the principal being changed is the `cpw --password=<new-pw-string>` parameter.  If you specify it as a parameter to cpw then you don't get prompted for it.

**Comment by macrotex** [comment]
The problem with using a command-line option is that it is hard to keep the command (and hence the password) out of system logs. That is why setting it via file is preferable.

**Comment by jaltman** [comment]
`[root@bethlehem ~]# cat > kfile
cpw --password=$$$$$$$$ bad.principal
[root@bethlehem ~]# kadmin -l < kfile
kadmin: cpw bad.principal: Principal does not exist
`
